include/json/value.h

```diff
@@ -392,6 +392,19 @@ Json::Value obj_value(Json::objectValue); // {}
   double asDouble() const;
   bool asBool() const;
 
+  template <typename T,
+            typename std::enable_if<std::is_enum<T>::value>::type* = nullptr>
+  T asEnum() {
+    switch (type()) {
+    case intValue:
+      return static_cast<T>(asLargestInt());
+    case uintValue:
+      return static_cast<T>(asLargestUInt());
```

Comment:
We don't really know that the LargestInt can be stored by enum T. We also don't know what what that would mean as an enum T. It could be invalid or have an unintended meaning. I think we should probably bow out and leave the conversion to enums to application-layer logic.

Now one way to do that would be to implement an ADL function protocol or some other kind of extension hook so that the enum maintainers can TELL jsoncpp what the appropriate value should be, but I don't think we've supported any other kinds of user-defined types this way, so I wouldn't want to start now.